src/Core/src/Credentials/WebIdentityProvider.php

```diff
@@ -128,14 +128,18 @@ private function getCredentialsFromRole(string $roleArn, string $tokenFile, ?str
      */
     private function getTokenFileContent(string $tokenFile): string
     {
-        if (!file_exists($tokenFile)) {
-            $tokenDir = \dirname($tokenFile);
-            $tokenLink = readlink($tokenFile);
-            clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . $tokenLink);
-            clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . \dirname($tokenLink));
-            clearstatcache(true, $tokenFile);
+        $token = @file_get_contents($tokenFile);
+
+        if (false !== $token) {
+            return $token;
         }
 
+        $tokenDir = \dirname($tokenFile);
+        $tokenLink = readlink($tokenFile);
+        clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . $tokenLink);
+        clearstatcache(true, $tokenDir . \DIRECTORY_SEPARATOR . \dirname($tokenLink));
+        clearstatcache(true, $tokenFile);
+
         if (false === $token = file_get_contents($tokenFile)) {
```

Comment:
shouldn't it be `@file_get_contents` here?

Reply:
I preferred to have the warning here anyway, since it helped me a lot debugging this regression.

IIRC, we get here only if we get the token file path from the env variable, so we shouldn't arrive to this point if there's no token to start with.